Backend/User/src/main/java/user/controller/ProfileController.java

```diff
@@ -0,0 +1,149 @@
+
+    @PostMapping("/register")
+    public ResponseEntity<Map<String, Object>> createProfile(@RequestBody ProfileRequest profileRequest) {
+        try {
+            boolean isCreated = profileService.createProfile(profileRequest);
+            if (isCreated) {
+                return ResponseManager.success("User registered successfully");
+            } else {
+                return ResponseManager.error(HttpStatus.INTERNAL_SERVER_ERROR, "Failed to register user");
+            }
+        } catch (IllegalArgumentException e) {
+            return ResponseManager.error(HttpStatus.BAD_REQUEST, e.getMessage());
+        } catch (Exception e) {
+            return ResponseManager.error(HttpStatus.INTERNAL_SERVER_ERROR, "Error: " + e.getMessage());
+        }
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<Map<String, Object>> authenticateUser(@RequestBody ProfileRequest loginRequest) {
+        try {
+            // Call the service to authenticate the user and get the JWT token
+            String jwtToken = profileService.authenticateUser(loginRequest);
+            
+            // Return a successful response with the JWT token
+            return ResponseManager.success("Success", Map.of(
+                "email", loginRequest.getEmail(),
+                "token", jwtToken
+            ));
+        } catch (IllegalArgumentException e) {
+            // Handle unauthorized errors
+            return ResponseManager.error(HttpStatus.UNAUTHORIZED, e.getMessage());
+        } catch (Exception e) {
+            // Handle other errors
+            return ResponseManager.error(HttpStatus.INTERNAL_SERVER_ERROR, "Error: " + e.getMessage());
+        }
+    }
+
+    @PostMapping("/logout")
+    public ResponseEntity<Map<String, Object>> logout(@RequestBody UUIDRequest uuidRequest) {
+        try {
+            profileService.logoutUser(uuidRequest.getUuid());
+            return ResponseManager.success("User logged out successfully.");
+        } catch (IllegalArgumentException e) {
+            return ResponseManager.error(HttpStatus.NOT_FOUND, "Error: " + e.getMessage());
+        } catch (Exception e) {
+            return ResponseManager.error(HttpStatus.INTERNAL_SERVER_ERROR, "Error: " + e.getMessage());
+        } 
+    }
+
+    @PostMapping("/profile")
+    public ResponseEntity<Map<String, Object>> getProfileByUUID(@RequestBody UUIDRequest uuidRequest) {
+        try {
+            User user = profileService.getProfileByUUID(uuidRequest.getUuid());
+            ProfileResponse profileResponse = new ProfileResponse(user);
+            return ResponseManager.success("Profile found", profileResponse);
+        } catch (UserNotFoundException e) {
+            return ResponseManager.error(HttpStatus.NOT_FOUND, "Error: " + e.getMessage());
+        } catch (Exception e) {
+            return ResponseManager.error(HttpStatus.INTERNAL_SERVER_ERROR, "Error: " + e.getMessage());
+        }
+    }
+
+    @PostMapping("/namelist")
```

Comment:
logic wise is ok, can consider `/profile/all/names-only` for controller naming consistency

Reply:
doners boss tq very much